### PR TITLE
[cmake] add build options for FTD, MTD, and RCP

### PR DESCRIPTION
**Pull request by jwhui**

### CODE REVIEW

File: examples/apps/ncp/CMakeLists.txt [hunk 1]
```diff
@@ -33,9 +33,16 @@ set(COMMON_INCLUDES
...
+    endif()
+    if(OT_MTD)
+        include(mtd.cmake)
+    endif()
 endif()
 
 if(OT_APP_RCP)
-    include(rcp.cmake)
+    if(OT_RCP)
+        include(rcp.cmake)
```
Comment (bukepo) [inline examples/apps/ncp/CMakeLists.txt:46]:
```suggestion
        include(rcp.cmake)
    else()
        message(FATAL_ERROR "OT_RCP is not enabled")
```

**Review by bukepo — APPROVED** [review]

**Review by bukepo — APPROVED** [review]
LGTM:+1:

**Review by Irving-cl — APPROVED** [review]
LGTM👍

**Review by abtink — APPROVED** [review]
LGTM. 👍

**Review by librasungirl — APPROVED** [review]

File: etc/cmake/options.cmake [hunk 2]
```diff
@@ -30,6 +30,10 @@ option(OT_APP_CLI "enable CLI app" ON)
 option(OT_APP_NCP "enable NCP app" ON)
 option(OT_APP_RCP "enable RCP app" ON)
 
+option(OT_FTD "enable FTD" ON)
+option(OT_MTD "enable FTD" ON)
```
Comment (wgtdkp) [inline etc/cmake/options.cmake:34]:
```suggestion
option(OT_MTD "enable MTD" ON)
```

File: examples/apps/cli/ftd.cmake [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+)
+
+target_include_directories(ot-cli-ftd PRIVATE ${COMMON_INCLUDES})
+
+target_link_libraries(ot-cli-ftd PRIVATE
+    openthread-cli-ftd
+    ${OT_PLATFORM_LIB}
+    openthread-ftd
+    ${OT_PLATFORM_LIB}
+    mbedcrypto
```
Comment (wgtdkp) [inline examples/apps/cli/ftd.cmake:40]: nit:
```suggestion
    ${OT_MBEDTLS}
```
Not changed by this PR, but I think we should use `${OT_MBEDTLS}` at here.

**Review by wgtdkp — APPROVED** [review]
LGTM 👍

**Pull request merged**